inttest/dualstack/dualstack_test.go

```diff
@@ -94,9 +111,122 @@ func (s *DualstackSuite) SetupSuite() {
+	s.Require().NoError(err)
+	s.NoError(common.WaitForPod(s.Context(), kc, "alpine-ipv6", "default"), "alpine pod did not start")
+
+	err = wait.PollImmediateWithContext(s.Context(), 100*time.Millisecond, time.Minute, func(ctx context.Context) (done bool, err error) {
+		s.Require().Equal(len(targetPodIPv6.Status.PodIPs), 2)
+		pod_ip := targetPodIPv6.Status.PodIPs[1].IP
+		target_ip := net.ParseIP(pod_ip)
+		s.Require().NotNil(target_ip)
+		out, err := common.PodExecCmdOutput(kc, restConfig, sourcePodIPv6.Name, sourcePodIPv6.Namespace, fmt.Sprintf("/usr/bin/wget -qO- %s", target_ip))
+		s.T().Log(out, err)
+		if err != nil {
+			return false, err
+		}
+		s.T().Log("server response", out)
+		return strings.Contains(out, "Welcome to nginx"), nil
+	})
+	s.Require().NoError(err)
+
+	// Test ipv4 connectivity between two pods
+	createdTargetPodIPv4, err := kc.CoreV1().Pods("default").Create(s.Context(), &corev1.Pod{
+		TypeMeta:   metav1.TypeMeta{Kind: "Pod", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: "nginx-ipv4"},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{{Name: "nginx-ipv4", Image: "docker.io/library/nginx:1.23.1-alpine"}},
+			NodeSelector: map[string]string{
+				"kubernetes.io/hostname": "worker0",
+			},
+		},
+	}, metav1.CreateOptions{})
+	s.Require().NoError(err)
+	s.Require().NoError(common.WaitForPod(s.Context(), kc, "nginx-ipv4", "default"), "nginx pod did not start")
+
+	targetPodIPv4, err := kc.CoreV1().Pods(createdTargetPodIPv4.Namespace).Get(s.Context(), createdTargetPodIPv4.Name, metav1.GetOptions{})
+	s.Require().NoError(err)
+
+	sourcePodIPv4, err := kc.CoreV1().Pods("default").Create(s.Context(), &corev1.Pod{
+		TypeMeta:   metav1.TypeMeta{Kind: "Pod", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: "alpine-ipv4"},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{{
+				Name:    "alpine",
+				Image:   "docker.io/library/alpine:" + getAlpineVersion(s.T()),
+				Command: []string{"sleep", "infinity"},
+			}},
+			NodeSelector: map[string]string{
+				"kubernetes.io/hostname": "worker1",
+			},
+		},
+	}, metav1.CreateOptions{})
+	s.Require().NoError(err)
+	s.NoError(common.WaitForPod(s.Context(), kc, "alpine-ipv4", "default"), "alpine pod did not start")
+
+	err = wait.PollImmediateWithContext(s.Context(), 100*time.Millisecond, time.Minute, func(ctx context.Context) (done bool, err error) {
+		s.Require().Equal(len(targetPodIPv4.Status.PodIPs), 2)
+		pod_ip := targetPodIPv4.Status.PodIPs[0].IP
+		target_ip := net.ParseIP(pod_ip)
+		s.Require().NotNil(target_ip)
+		out, err := common.PodExecCmdOutput(kc, restConfig, sourcePodIPv4.Name, sourcePodIPv4.Namespace, fmt.Sprintf("/usr/bin/wget -qO- %s", target_ip))
+		s.T().Log(out, err)
+		if err != nil {
+			return false, err
+		}
+		s.T().Log("server response", out)
+		return strings.Contains(out, "Welcome to nginx"), nil
+	})
+	s.Require().NoError(err)
+
 	s.client = client
 }
 
+func getAlpineVersion(t *testing.T) string {
```

Comment:
You can remove all this logic if you use nginx for both pods. The nginx image has both wget and curl so you can remove this.

Reply:
I did as you suggested, see new commit.

Reply:
Looks better, we need to do change some stuff on the CI before merging. Other than the linting stuff that Tom mentioned it looks good